src/core/thread/mesh_forwarder_ftd.cpp

```diff
@@ -751,46 +751,47 @@ void MeshForwarder::UpdateFragmentPriority(Lowpan::FragmentHeader &aFragmentHead
                                            uint16_t                aSrcRloc16,
                                            Message::Priority       aPriority)
 {
-    FragmentPriorityEntry *entry;
+    FragmentPriorityList::Entry *entry;
```

Comment:
Adding a note on the change here:

This commit changes the order the check.

- We first check if there is an existing entry with a matching source RLOC16 and tag.
   - This handles the case where we may receive duplicate first fragments (this can, for example, happen with MLE msgs since they don't use MAC layer security and therefore won't be filtered as duplicate by `Mac`).
- Only if there is no existing entry, we create a new one when datagram offset is zero (i.e., it's a first frag).